lib/src/util/dart_type_utilities.dart

```diff
@@ -49,6 +49,69 @@ class DartTypeUtilities {
     return null;
   }
 
+  /// Return whether the canonical elements of two elements are equal.
+  static bool canonicalElementsAreEqual(Element element1, Element element2) =>
+      getCanonicalElement(element1) == getCanonicalElement(element2);
+
+  /// Returns whether the canonical elements from two nodes are equal.
+  ///
+  /// As in, [getCanonicalElementFromIdentifier], the two nodes must be
+  /// [Expression]s in order to be compared (otherwise `false` is returned).
+  ///
+  /// The two nodes must both be a [SimpleIdentifier], [PrefixedIdentifier], or
+  /// [PropertyAccess] (otherwise `false` is returned).
+  ///
+  /// If the two nodes are PrefixedIdentifiers, or PropertyAccess nodes, then
+  /// `true` is returned only if their canonical elements are equal, in
+  /// addition to their prefixes' and targets' (respectfully) canonical
+  /// elements.
+  ///
+  /// There is an inherent assumption about pure getters. For example:
+  ///
+  ///     A a1 = ...
+  ///     A a2 = ...
+  ///     a1.b.c; // statement 1
+  ///     a2.b.c; // statement 2
+  ///     a1.b.c; // statement 3
+  ///
+  /// The canonical elements from statements 1 and 2 are different, because a1
+  /// is not the same element as a2.  The canonical elements from statements 1
+  /// and 3 are considered to be equal, even though `A.b` may have side effects
+  /// which alter the returned value.
+  static bool canonicalElementsFromIdentifiersAreEqual(
+      AstNode rawNode1, AstNode rawNode2) {
```

Comment:
Consider dropping the prefix "raw" from these. I don't think it adds to the semantic value.